compiler/src/codegen/emitmod.re

```diff
@@ -1,8 +1,72 @@
 open Grain_typed;
 open Grain_utils;
-open Compmod;
+open Mashtree;
 
-let emit_module = ({asm, signature}, outfile) => {
+type bad_object =
+  | InvalidMagic
+  | InvalidVersion
+  | Corrupted;
+
+exception BadObject(bad_object);
+
+let emit_object = (mashed: mash_program, outfile) => {
```

Comment:
Could we add a small little description of what the object files look like either here or in the contributer docs. Something like whats in the pr description.

Reply:
Done.